src/transformers/utils/quantization_config.py

```diff
@@ -178,6 +185,83 @@ def update(self, **kwargs):
         return unused_kwargs
 
 
+@dataclass
+class HQQConfig(QuantizationConfigMixin):
+    """
+    Main HQQConfig.
+
+    Args:
+        quant_config (`HQQBaseQuantizeConfig, Dict[HQQBaseQuantizeConfig]`):
+            Dictionaries used for detailed quantization config. If it's a single HQQBaseQuantizeConfig, linear layers will the same config.
+            If the dictionary contains linear tags, each linear tag will use the corresponding config
+        skip_modules (List[str])::
+            nn,Linear layers to skip.
+        show_progress (bool):
+            Show tqdm quantization for each shard
+        kwargs (`Dict[str, Any]`):
+            Additional parameters from which to initialize the configuration object.
+    """
+
+    def __init__(
+        self,
+        quant_config=hqq_default_config,
```

Comment:
For transformers integration, I don't know if it is interesting to allow users to quantize each layer in a different setting. This makes the `quant_config` arg quite complex to use too + we also need to pass `HQQBaseQuantizeConfig` a object and it is best if we don't depend on that. 
I suggest that we just pass the args of `HQQBaseQuantizeConfig` in `HQQConfig` (bits, group_size, offload_meta). We could potentially add support for linear tags in another argument such as `quant_config_dict`.  WDYT ? cc @amyeroberts if you have any recommendation

Reply:
Quantizing each layer with separate configs was a request by many users, that's why we added it. It is particularly useful for larger models, especially MoEs where experts are more heavily quantized than attention modules. 
Also, some parameters are offloaded to the CPU, and if you offload all of them with a single parameter it becomes too slow. 
My recommendation is to keep them, but simplify the arguments. We can get rid of ```HQQBaseQuantizeConfig```. For something like Llama or Mixtral, it just a max of 7 lines of config data